Podcast/UserDetailViewController.swift

```diff
@@ -0,0 +1,413 @@
+    let subscriptionsHeaderViewTag = 3110
+    
+    var scrollYOffset: CGFloat = 118
+    
+    var profileTableView: UITableView!
+    var userDetailHeaderView: UserDetailHeaderView!
+    var navBar: UserDetailNavigationBar!
+    var isLoading: Bool = true
+    
+    var user: User!
+    var subscriptions: [Series] = []
+    var recasts: [Episode] = []
+    
+    var currentlyPlayingIndexPath: IndexPath?
+    
+    convenience init(user: User) {
+        self.init()
+        self.user = user
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Do any additional setup after loading the view.
+        profileTableView = UITableView(frame: .zero, style: .grouped)
+        profileTableView.register(EpisodeTableViewCell.self, forCellReuseIdentifier: episodeCellReuseId)
+        profileTableView.register(NullProfileTableViewCell.self, forCellReuseIdentifier: nullEpisodeCellReuseId)
+        profileTableView.register(UserDetailSectionHeaderView.self, forHeaderFooterViewReuseIdentifier: headerViewReuseId)
+        profileTableView.delegate = self
+        profileTableView.dataSource = self
+        profileTableView.backgroundColor = .paleGrey
+        profileTableView.rowHeight = UITableViewAutomaticDimension
+        profileTableView.separatorStyle = .none
+        profileTableView.showsVerticalScrollIndicator = false
+        profileTableView.contentInsetAdjustmentBehavior = .never
+        profileTableView.contentInset.top = 0
+        mainScrollView = profileTableView
+        view.addSubview(profileTableView)
+        profileTableView.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+        
+        userDetailHeaderView = UserDetailHeaderView(frame: CGRect(x: 0, y: 0, width: view.frame.width, height: UserDetailHeaderView.minHeight))
+        profileTableView.tableHeaderView = userDetailHeaderView
+        userDetailHeaderView.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+            make.height.greaterThanOrEqualTo(UserDetailHeaderView.minHeight)
+            make.width.equalToSuperview()
+        }
+        userDetailHeaderView.delegate = self
+        userDetailHeaderView.subscriptionsHeaderView.delegate = self
+        userDetailHeaderView.subscriptionsHeaderView.tag = subscriptionsHeaderViewTag
+        userDetailHeaderView.subscriptionsView.register(SeriesGridCollectionViewCell.self, forCellWithReuseIdentifier: seriesCellReuseId)
+        userDetailHeaderView.subscriptionsView.register(NullProfileCollectionViewCell.self, forCellWithReuseIdentifier: nullSeriesCellReuseId)
+        userDetailHeaderView.subscriptionsView.delegate = self
+        userDetailHeaderView.subscriptionsView.dataSource = self
+        userDetailHeaderView.setSubscriptions(hidden: isLoading)
+        userDetailHeaderView.configure(for: user, isMe: (user == System.currentUser!)) // Safe unwrap, guaranteed to be there
+        
+        navBar = UserDetailNavigationBar()
```

Comment:
Can we safely unwrap `(navigationController?.navigationBar.frame.height)!`, etc.

Reply:
This is always safe.